receiver/otlpreceiver/otlp.go

```diff
@@ -186,8 +187,18 @@ func (r *otlpReceiver) Start(ctx context.Context, host component.Host) error {
 func (r *otlpReceiver) Shutdown(ctx context.Context) error {
 	var err error
 
+	if r.cfg != nil {
+		if r.cfg.HTTP != nil && r.cfg.HTTP.TLSSetting != nil {
+			err = r.cfg.HTTP.TLSSetting.Shutdown()
+		}
+
+		if r.cfg.GRPC != nil && r.cfg.GRPC.TLSSetting != nil {
+			err = multierr.Append(err, r.cfg.GRPC.TLSSetting.Shutdown())
+		}
+	}
```

Comment:
Looking at this some more I really don't like that any component that uses `configtls` has to add this check.  I am worried about components not realizing that the `configtls.TLSSetting` embedded in `confighttp.ServerConfig` needs a `Shutdown` call.

I think it is out of scope of this PR, which is only trying to solve a leak added by existing functionality`, but I think we should take another look at how `config` modules should handle goroutines and `Shutdown`.